src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -503,13 +503,15 @@ class ResourceService(
         case LoadResourceAuthDomainResult.ResourceNotFound => IO.raiseError(
           new WorkbenchExceptionWithErrorReport(
             ErrorReport(StatusCodes.BadRequest, s"ResourceId ${policyId.resource} not found.")))
-        case LoadResourceAuthDomainResult.Constrained(_) if public =>
+        case LoadResourceAuthDomainResult.Constrained(_) =>
           // resources with auth domains logically can't have public policies but also technically allowing them poses a problem
           // because the logic for public resources is different. However, sharing with the auth domain should have the
           // exact same effect as making a policy public: anyone in the auth domain can access.
+          if (public)
           IO.raiseError(
             new WorkbenchExceptionWithErrorReport(
               ErrorReport(StatusCodes.BadRequest, "Cannot make auth domain protected resources public. Share directly with auth domain groups instead.")))
+          else IO.unit //TODO: non exhaustive matching bug. What should happen in this case?
```

Comment:
`IO.unit` seems appropriate here, the resource is constrained by an auth domain, but `public` is `false` which is what it already will be, so fine to just do nothing in my opinion